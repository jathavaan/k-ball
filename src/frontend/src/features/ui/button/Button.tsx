import { StyledButton } from "./button.style";
import { ButtonProps } from "./button.types";
import { CircularProgress } from "@mui/material";

export const Button = ({ text, isLoading = false, ...props }: ButtonProps) => {
  return (
    <StyledButton {...props}>
      {!isLoading ? text : <CircularProgress color="inherit" size={25} />}
    </StyledButton>
  );
};
