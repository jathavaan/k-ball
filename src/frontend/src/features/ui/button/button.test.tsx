import { render } from "@testing-library/react";
import { Button } from "./Button";

it("should match Button snapshot", () => {
  const { asFragment } = render(
    <Button
      onClick={function (): void {
        throw new Error("Function not implemented.");
      }}
      text={""}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
