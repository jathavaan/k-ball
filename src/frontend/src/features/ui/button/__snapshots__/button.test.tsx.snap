// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match Button snapshot 1`] = `
<DocumentFragment>
  <button
    class=MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-dynamic-MuiButtonBase-root-MuiButton-root
    tabindex="0"
    type="button"
  />
</DocumentFragment>
`;
